### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by kensteph**
In this PR I : 
- Install axios.
- Create getBookFromAPPi function.
- Configure extra Reducers to manage lifecycle actions.
- Fetch the books from Bookstore API when the component has rendered with useEffect.
- Add postBookToAPI function.
- Add removeBookFromAPI function

### CODE REVIEW

File: src/redux/features/books/booksSlice.js [hunk 1]
```diff
@@ -1,42 +1,104 @@
...
-  {
-    id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
+const baseURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/xOI7HhKVUsDCTkv7qbXd/books';
+
+// =============== Asynchronous =============
+
+export const getBookFromAPI = createAsyncThunk(
```
Comment (FrederickMih) [inline src/redux/features/books/booksSlice.js:9]: - [ ] Hi @kensteph, it seems the getBookFromAPI function was not dispatched in your BookList component. As requested in the requirements of this project lets us kindly avoid unwanted component renderings by using the useEffect Hook to dispatch the getBookFromAPI function. **Hint**

```js
import { fetchBooks } from "................."

function BookList() {
  const dispatch = useDispatch()


  useEffect(() => {
    dispatch(getBookFromAPI())
  }, [])

  return <div>{/* BookList JSX including books data */}</div>
}


```

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @kensteph , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Great commit history :+1: 
- The add book and remove book features still works the same :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kensteph** [comment]
Thank you @FrederickMih  for your review.

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @kensteph, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- Linter checks passed :heavy_check_mark: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**